### DISCUSSION TITLE
Sonoff ZBDongle-P update firmware

### DISCUSSION THREAD

**Discussion by titou4307** [Koenkk/zigbee2mqtt#16556] [Devices]
Hello at all

I'am new on zigbee2mqtt so I discover many thing on each day ....

My Z2M has coordinator Sonoff ZBDongle-P and work perfectly
For this in the past whan I received this USB Dongle I have flashed this with the firmware for 20220219

Now I have about 30 zigbee controllers paired.....many are into the wall near switch for light and need for access to unmount the switch....

There  a new firmware out : 20221226

If I flash the new firmware does it needeed to pair all the controllers (the 30 unity !!) or it not needeed ??

Many thanks

**Comment by Matriciel** [comment]
Hi,
It is not needed if you only flash the dongle and don't change any addresses.

**Comment by titou4307** [comment]
@Matriciel 
Thanks for your response, il will flash this dongle with the last firmware

**Reply by notabenex** [reply]
Hello, did it worked correctly?

**Comment by titou4307** [comment]
I have not done the flash (need to find a little time.....)
I will made a return here after

**Comment by satter** [comment]
Just did a firmware upgrade for my Sonoff Zigbee Dongle Plus from 20211217 to 20221226  
Everything seems to be working fine, not a single device dropped off, no re-pairing required  

I've stopped z2m and homebridge services and then performed upgrade:  
```
~# python3 /opt/cc2538-bsl/cc2538-bsl.py -e -v -w --bootloader-sonoff-usb /tmp/CC1352P2_CC2652P_launchpad_coordinator_20221226.hex
sonoff
Opening port /dev/ttyUSB0, baud 500000
Reading data from /tmp/CC1352P2_CC2652P_launchpad_coordinator_20221226.hex
Firmware file: Intel Hex
Connecting to target...
CC1350 PG2.0 (7x7mm): 352KB Flash, 20KB SRAM, CCFG.BL_CONFIG at 0x00057FD8
Primary IEEE Address: 00:12:4B:00:25:8D:3C:29
    Performing mass erase
Erasing all main bank flash sectors
    Erase done
Writing 360448 bytes starting at address 0x00000000
Write 104 bytes at 0x00057F980
    Write done
Verifying by comparing CRC32 calculations.
    Verified (match: 0xa9dc145d)
```

**Reply by florinssk** [reply]
do you have a tutorial for this ?

**Reply by chris-1243** [reply]
You may check this [link](https://github.com/JelmerT/cc2538-bsl).

You should find the python version as well as a docker version (which I used and I am pretty happy with it).

Everything required is explained
